rfc/1216-resolve-cloud-tokens.md

```diff
@@ -0,0 +1,845 @@
+In this case, the implementation will return the *physical id* as returned by 
+the [cloudformation/DescribeStackResource][2] API, assuming they represent the same 
+thing. See [Appendix](#ref-vs-physicalresourceid) for more details and research on this.
+
+- The implementation uses the *physical id* as the `Identifier` argument when calling 
+the [cloudcontrol/GetResource][1] API, assuming they map 1:1. See [Appendix](#physicalresourceid-vs-primaryidentifier) 
+for more details and research on this.
+
+#### Package `@cdk8s/cdktf-token-resolver`
+
+> PoC: [cdktf-resolver.ts](./../examples/typescript/resolve-cloud-tokens/cdktf-resolver.ts)
+
+A new jsii package containing a class that implements the `IResolver` interface. It can 
+identify CDKTF tokens, and fetch their concrete values by issuing terraform state calls.
+
+Following is an outline of the implementation:
+
+```mermaid
+flowchart TD
+    A("CdkTfResolver.resolve")-->|not Token.isUnresolved|B("Return")
+    A("CdkTfResolver.resolve")-->|Token.isUnresolved|D("cloudformation.DescribeStackResource")
+    D("terraform show -json")-->E("Find Resource By Address")
+    E("Find Resource By Address")-->F("Extract Resource Properties")
+    F("Extract Resource Properties")-->G("Return Attribute Value")
+```
+
+### Is this a breaking change?
+
+No
+
+### What alternative solutions did you consider?
+
+#### Outputs
+
+In this solution, whenever cdk8s encounters an AWS CDK token, instead of trying to 
+fetch its corresponding *attribute* value, it will fetch its corresponding *output* value.
+If the value doesn't exist, cdk8s will add a corresponding `CfnOutput` resource to the 
+stack. It would look something like this:
+
+```ts
+// some output id generated from the token (after stack.resolve)
+const outputId = 'some-stable-id'
+
+try {
+  return fetchOutputValue(stack, outputId)
+} catch (error: OutputNotFound) {
+  new CfnOutput(stack, outputId, { value: bucket.bucketName });
+  // nothing else we can return here...
+  return bucket.bucketName
+}
+```
+
+This solution has the benefit of automatically supporting every value that 
+can be defined in an AWS CDK application. It also simplifies the fetching logic because 
+it requires a single `cloudformation.DescribeStacks` call. 
+
+However, this option was discarded because of usability concerns:
+
+- Requires synthesis of the cdk8s app to happen before the AWS CDK app. This can only be 
+controlled by the user, and therefore error prone.
+- Synthesizing the AWS CDK application separately from the cdk8s application will result in a different cloud assembly.
+- The cdk8s application needs to be synthesized twice, once to add the CloudFormation outputs, and once to fetch their values. 
+This means that the first synthesis will inherently produce an invalid manifest.
+- Will create a cyclic dependency in case the cdk8s and AWS CDK application are in different repositories / packages.
```

Comment:
I don't get this concern either. Why? The CDK app doesn't need to know about the identity of its consumer, so looks to me the dependency arrow only points one way?